File: dotnet/src/Envdoctor/Commands/InitCommand.cs
using Envdoctor.Generators;

namespace Envdoctor.Commands;

public sealed class InitArgs
{
    public string? Root { get; set; }
    public bool Force { get; set; }
}

public static class InitCommand
{
    public static int Run(InitArgs args)
    {
        var root = Path.GetFullPath(args.Root ?? ".");

        var configPath = Path.Combine(root, "envdoctor.config.toml");

        if (File.Exists(configPath) && !args.Force)
        {
            Console.Error.WriteLine($"Config already exists at {configPath}. Use --force to overwrite.");
            return 1;
        }

        var template = SchemaGenerator.GenerateConfigTemplate();
        File.WriteAllText(configPath, template);

        Console.Out.WriteLine($"Created {configPath}");
        Console.Out.WriteLine("Edit the file to customize your configuration, then run `envdoctor scan`.");

        return 0;
    }
}

File: swift/Sources/EnvdoctorKit/Commands/InitCommand.swift
import Foundation

public struct InitArgs {
    public var root: String?
    public var force = false

    public init() {}
}

public enum InitCommand {
    /// `envdoctor init` — bootstrap a config for the project. Non-destructive:
    /// an existing config is only overwritten with `--force`.
    public static func run(_ args: InitArgs) throws -> Int {
        let root = resolveRoot(args.root)
        let configPath = root + "/envdoctor.config.toml"

        if FileManager.default.fileExists(atPath: configPath), !args.force {
            FileHandle.standardError.write("Config already exists at \(configPath). Use --force to overwrite.\n")
            return 1
        }

        let template = SchemaGenerator.generateConfigTemplate()
        try template.write(toFile: configPath, atomically: true, encoding: .utf8)

        print("Created \(configPath)")
        print("Edit the file to customize your configuration, then run `envdoctor scan`.")

        return 0
    }
}
